docs/recipes/precompiling-with-webpack.md

```diff
@@ -29,3 +37,226 @@ module.exports = {
+
+```js
+// Before
+import fresh from '../src';
+// After
+import fresh from '../_src';
+```
+
+#### Single entry file
+To pre-compile folder with multiple test files, we can use globbing(see [stackoverflow answer](http://stackoverflow.com/questions/32874025/how-to-add-wildcard-mapping-in-entry-of-webpack/34545812#34545812)). Although this performs quite possibly the best, it comes at a cost. AVA uses the filename and path for test names in verbose mode as well as test failure reports. This can make the error report harder to understand. Also, it can matter if tests modify globals, Node.js built-ins. You wouldn't want that to leak across tests.
+
+###### webpack.config.js
+
+```js
+const path = require('path');
+const glob = require('glob');
+
+const nodeExternals = require('webpack-node-externals');
+
+module.exports = {
+	target: 'node',
+	entry: glob.sync('./test/**/*.js'),
+	output: {
+		path: path.resolve(__dirname, '_build'),
+		filename: 'tests.js'
+	},
+	externals: [nodeExternals()],
+	module: {
+		rules: [{
+			test: /\.(js|jsx)$/,
+			use: {
+				loader: 'babel-loader',
+				options: { cacheDirectory: true }
+			}
+		}]
+	}
+};
+```
+
+<details>
+<summary>Error report comparison</summary>
+
+```
+# Before
+  aggregations-test » cardinality-agg » sets precision_threshold option
+  E:\Projects\repos\elastic-builder\test\_macros.js:167
+
+   166:         const expected = getExpected(keyName, recursiveToJSON(propValue));
+   167:         t.deepEqual(value, expected);
+   168:     }
+
+  Difference:
+
+      Object {
+        my_agg: Object {
+          cardinality: Object {
+    -       precision_threshol: 5000,
+    +       precision_threshold: 5000,
+          },
+        },
+      }
+
+# After
+  sets precision_threshold option
+  E:\Projects\repos\elastic-builder\_build\tests.js:106
+
+   105:                     column: 21
+   106:                 }
+   107:             },
+
+  Difference:
+
+      Object {
+        my_agg: Object {
+          cardinality: Object {
+    -       precision_threshol: 5000,
+    +       precision_threshold: 5000,
+          },
+        },
+      }
+
+```
+
+</details>
+
+#### Multiple entry files
+We can ask webpack to generate multiple entry files. This helps retain file names so that error reports are easy to interpret. But each entry file gets it's own copy of the source files. This results in considerably larger file sizes. This can [perform quite poorly](https://github.com/avajs/ava/pull/1385#issuecomment-304684047) on the first execution.
+
+###### webpack.config.js
+
+```js
+const path = require('path');
+const glob = require('glob');
+
+const nodeExternals = require('webpack-node-externals');
+
+const testFiles = glob.sync('./test/**/*.js');
+
+const entryObj = {};
+const len = testFiles.length;
+for (let idx = 0; idx < len; idx++) {
+	const file = testFiles[idx];
+	entryObj[path.basename(file, path.extname(file))] = file;
+}
```

Comment:
I too used `reduce` initially but felt the `for` method was easier to understand. I'll change it to use `reduce`.